cerberus/widgets.py

```diff
@@ -0,0 +1,172 @@
+# Standard Library
+from collections.abc import Callable
+from typing import Any
+
+# Django
+from django import forms
+from django.forms import widgets
+
+# Third Party
+from djmoney.forms import MoneyWidget
+
+# Locals
+from .utils import rgetattr
+
+
+class SingleMoneyWidget(MoneyWidget):
+    def __init__(self, attrs=None, *args, **kwargs):
+        attrs = attrs or {}
+        attrs.update({"step": "any", "min": "0"})
+
+        super().__init__(
+            amount_widget=forms.NumberInput(attrs=attrs),  # type: ignore
+            currency_widget=forms.HiddenInput(),
+            *args,
+            **kwargs,
+        )
+
+    def id_for_label(self, id_):
+        return f"{id_}_0"
+
+
+Attr_callback = Callable[[str, Any, int | str, dict[str, Any]], dict[str, Any] | None]
+
+
+class OptionAttrs(widgets.ChoiceWidget):
+    attr_callback: Attr_callback | None
+
+    def __init__(self, attr_callback: Attr_callback | None = None, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.attr_callback = attr_callback
+
+    def create_option(
+        self,
+        name: str,
+        value: Any,
+        label: int | str,
+        selected: set[str] | bool,
+        index: int,
+        subindex: int | None = None,
+        attrs: dict[str, Any] | None = None,
+    ) -> dict[str, Any]:
+        option = super().create_option(name, value, label, selected, index, subindex, attrs)
+        if self.attr_callback:
+            option["attrs"] = self.attr_callback(name, value, label, option["attrs"])
+
+        return option
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (performance):** Consider optimizing CheckboxTable's create_option method for large datasets.

The current implementation iterates over all model fields for each option, which might be inefficient for large datasets or complex models. Consider caching or pre-fetching data if possible.